lib/handleRequest.js

```diff
@@ -125,28 +129,64 @@ function validationCompleted (request, reply, validationErr) {
 function preHandlerCallback (err, request, reply) {
   if (reply.sent) return
 
-  if (err != null) {
-    reply[kReplyIsError] = true
-    reply.send(err)
-    return
+  const context = request[kRouteContext]
+
+  const store = {
+    request,
+    reply,
+    async: false,
+    route: {
+      url: context.config.url,
+      method: context.config.method
+    }
   }
 
-  let result
+  if (context[kFourOhFourContext] === null) {
```

Comment:
@mcollina I was able to fix the issue with 404 routes executing twice with this check. But I'm wondering if the solution is now too-specific to handling 404 errors and if other situations can trigger the same problem?

Reply:
Possibly the error path, but I think you handled that already.